contracts/Vault.sol

```diff
@@ -526,12 +526,17 @@ contract Vault is IVault {
             _token == token &&
             _balance < _tokenBalance
         ) {
-            uint256 _redundant;
-            unchecked{
-                _redundant = _tokenBalance - _balance;
+            uint256 _utilized = controller.valueAll();
+            uint256 _actualValue = IERC20(token).balanceOf(address(this)) + _utilized;
```

Comment:
can we use _tokenBalance and _balance instead of re-doing IERC20(token).balanceOf(address(this))?